src/proxy/socks5.rs

```diff
@@ -81,7 +81,8 @@ impl Socks5 {
                 // but ProxyInfo is overloaded and only `outbound` should ever use the pool.
                 let pool = crate::proxy::pool::WorkloadHBONEPool::new(
                     self.pi.cfg.clone(),
-                    self.enable_orig_src,
+                    // Do not need to spoof with inpod mode for outbound
+                    self.enable_orig_src && !self.pi.cfg.inpod_enabled,
```

Comment:
Ditto - hoist the config check to `new` and just turn it off at the beginning.

Reply:
:+1: